src/preprocess.rkt

```diff
@@ -26,17 +26,21 @@
   (define sorted (sort (map list indicies values) (lambda (a b) (< (first a) (first b)))))
   (list-set-multiple point sorted))
 
-(define (sort-group variables point preprocess repr)
-  (match-define (list 'sort vars ...) preprocess)
-  (apply-to-group variables point vars (lambda (group) (sort group (curryr </total repr)))))
-
 (define (apply-preprocess sampled-point preprocess-structs ctx)
   (cond
     [(empty? preprocess-structs)
      sampled-point]
-    ;; Add more preprocess cases here- for now, only symmetry-group exists
     [else
-     (define pt* (sort-group (context-vars ctx) sampled-point (first preprocess-structs) (context-repr ctx)))
+     (define variables (context-vars ctx))
+     (define repr (context-repr ctx))
+     (define pt*
+       (match (first preprocess-structs)
+         [(list 'sort vars ...)
+          (apply-to-group variables sampled-point vars (lambda (group) (sort group (curryr </total repr))))]
+         [(list 'abs var)
+          ;; TODO: Big question, is there more than one variable ever? Is any of this even right?
+          ;; I suppose we could map abs on a list of one variable but that feels sketchy
+          (apply-to-group variables sampled-point (list var) (match-lambda [(list x) (list (abs x))]))]))
```

Comment:
This is about right, but if you want to be fully correct you'll want to replace `abs` with `(operator-info (get-parametric-operator 'fabs repr) 'fl)`. Here `get-parameteric-operator` converts `'fabs` to `'fabs.f64` or whatever representation you have, and the `operator-info` then turns it into an executable Racket operation. I suppose you'd also want to handle the case of a representation that doesn't supply `fabs`; not sure where exactly to do that but at least let's leave a TODO.